sui-execution/v0/sui-adapter/src/execution_engine.rs

```diff
@@ -440,6 +440,67 @@ mod checked {
         }
     }
 
+    #[instrument(
+        name = "commit_transaction_without_execution",
+        level = "debug",
+        skip_all
+    )]
+    pub fn commit_transaction_without_execution<Mode: ExecutionMode>(
```

Comment:
Can't we just give an invariant violation for v0? This code path shouldn't be executed in v0 right?